vpd-tool/src/vpd_tool.cpp

```diff
@@ -85,4 +86,47 @@ int VpdTool::readKeyword(const std::string& i_vpdPath,
     }
     return l_rc;
 }
+
+int VpdTool::writeKeyword(const std::string& i_vpdPath,
+                          const std::string& i_recordName,
+                          const std::string& i_keywordName,
+                          const std::string& i_keywordValue,
+                          const bool i_onHardware,
+                          const std::string& i_filePath) noexcept
+{
+    int l_rc = constants::FAILURE;
+    try
+    {
+        if (i_vpdPath.empty() || i_recordName.empty() || i_keywordName.empty())
+        {
+            throw std::runtime_error("Received input is empty.");
+        }
+
+        if (!i_filePath.empty())
+        {
+            // ToDo: Take keyword value from the file
+        }
+
+        std::string l_vpdPath{i_vpdPath};
```

Comment:
If you have to change this in the body, why do you have to take it as const reference?

Reply:
user gives object path from /system similar to the existing style.
so am keeping user input as const ref and modifying the path by appending initial base path ("/xyz/openbmc_project/inventory")

Reply:
I know what you are doing, I want to understand why?

What is the point taking a parameter as constant reference and then copying to a non const variable and then modify it.